### PR TITLE
Make Cloud VPN setup contextual and accessible in Ports

### LINKED ISSUE #12489: Make Cloud Ports VPN guidance obvious and contextual
## Problem

The Cloud Machines right sidebar currently shows a small orange caution icon beside Ports and a prominent “Set Up cmux VPN…” banner at the top. The icon is not obviously clickable, and the explanation is not visible where the user is looking. The top banner makes VPN look like a prerequisite for ordinary Cloud browsing even though cmux's built-in port forwarding and Desktop path work without the system-wide VPN.

## Requested behavior

- Remove the standalone “Set Up cmux VPN…” banner from the top of the Cloud/Machines sidebar.
- Keep the Ports section as the contextual place for VPN guidance. When Cloud VPN is off and the Ports section has no reachable ports, show a clear nested callout under “No reachable ports” with an explicit message that cmux's built-in forwarding still works and Cloud VPN is optional for access from Safari, Chrome, and other apps.
- Make the caution affordance visibly interactive: use a recognizable warning/help button or labeled row, a sufficient hit target, hover tooltip, keyboard/accessibility label, and a “Learn more” or “Set Up Cloud VPN” action.
- Clicking the action opens the existing Cloud VPN setup pane in cmux, preserving the current workspace and focus behavior. Do not open a generic top-level setup banner or imply that VPN is required to open Desktop/Ports in cmux.
- Preserve actionable details for the cases where Ports are present, unavailable, loading, or the Cloud link is disconnected. Avoid covering the machine/workspace tree.
- Keep the explanation localized and consistent with the existing Cloud VPN setup copy.

## Acceptance criteria

1. No standalone VPN setup banner appears at the top of the Machines/Cloud sidebar.
2. With Cloud VPN disabled and no reachable ports, the Ports section contains an obvious, labeled, hoverable/keyboard-accessible callout under the empty state.
3. The callout explains optional system-wide VPN access versus cmux's built-in forwarding in plain language.
4. Selecting “Set Up Cloud VPN” or the warning affordance opens the existing setup pane without changing the user's selected workspace unexpectedly.
5. Ports with reachable entries remain compact and useful; the guidance does not obscure port rows.
6. Add focused projection/UI behavior tests for banner removal, empty Ports callout, hover/action state, and setup-pane routing. Do not touch Swift warning/length budget TSVs.

Related: #12407, #12417, #12431, #12093.

### PR DESCRIPTION
**Pull request by austinywang**
Ports now has a gray **(?)** help button that opens the existing Cloud VPN setup pane directly. Hover explains that cmux’s in-app forwarding works without a system VPN, while Cloud VPN lets Safari, Chrome, and other apps open private VM ports.

With the supported VPN off and a successful empty port scan, the section shows **Configure cmux VPN** and a wrapping explanation in place of “No reachable ports.” Reachable rows stay compact, and loading, disconnected, unknown, and unsupported states keep their existing presentation. The standalone setup banner above the machine tree is removed; active-operation and plan notices remain.

Setup uses the existing dedicated guide workspace in the originating window, reuses an open guide, and honors background focus requests. Keeping the guide separate preserves Cloud workspace layout synchronization, which pauses when local utility panels are mixed into a machine workspace. Both native button presentations share one activation path, localized hover/accessibility help, stable identifiers, and disabled-state handling. The help button has a 28×24 hit target.

Resize invalidation caches successful-empty-Ports node identities during snapshot application, including content-only adoption. Each resize resolves only those nodes to their current visible rows. Tests cover large trees, collapse/expansion, retained item identity, disconnection, replacement, and VPN state changes.

Fixes #12489.

Validation:

- [Tagged hosted build](https://github.com/manaflow-ai/cmux/actions/runs/34755825524) — passed on `c6b380de94`; installed tag `cloud-vpn-12489` verifies its code signature.
- [Affordance, activation, and setup routing tests](https://github.com/manaflow-ai/cmux/actions/runs/34755825487) — 10 tests passed on `c6b380de94`. The complete unit-target build reports no errors or warnings in changed Swift files.
- [Resize regression tests](https://github.com/manaflow-ai/cmux/actions/runs/34754539463) — 3 tests passed.
- [Banner and localized-help tests](https://github.com/manaflow-ai/cmux/actions/runs/34754539608) — 7 tests passed.
- [Pre-fix resize regression checkpoint](https://github.com/manaflow-ai/cmux/actions/runs/34754522619) — failed at the intended assertions: 5 resizes scanned 5,005 items with 1,000 unrelated rows. The checkpoint includes the same prerequisite test repairs. Regression tests precede the VPN fix in separate commits.
- The [Ports context-menu suite](https://github.com/manaflow-ai/cmux/actions/runs/34749022165) passed.
- File-length budgets, project normalization, test wiring, diff formatting, and the nine-language localization audit pass. Both new keys also have translated entries in all 20 catalog languages. Neither Swift budget TSV changed.

Merged `origin/main` and resolved the test conflict and duplicate extracted terminal methods. Main’s terminal repairs supersede the earlier prerequisite commits; the final application diff is scoped to VPN guidance. Hosted compilation also exposed malformed assertions in three tests recently added on main: a missing `try`, a nested `#require`, and mutating readiness checks inside `#expect`. Those test-only repairs are included so focused validation can execute; the cancellation test moves to a small extension file to preserve its original file’s length budget.

The final test-only cleanup replaces a deprecated workspace-creation API in the affordance fixture; the tagged build and affected suite passed on `c6b380de944fc66548c51d824003b49eb9a850f4`. The unchanged resize and banner suites passed on `2b1b5a4a40`. The application also passed fleet launch and debug-socket verification.

Validation limits: no local xcodebuild or XCUITests were run. Prior isolated fleet screenshots and accessibility snapshots verified the controls; full authenticated Cloud dogfooding remains unavailable because the shared development backend hostname does not resolve. The fleet recording attempt failed, so there is no demo video.

### CODE REVIEW

File: cmuxTests/CloudTunnelBannerTests.swift [hunk 1]
```diff
@@ -15,6 +15,13 @@ struct CloudTunnelBannerTests {
     private static let extensionID = "com.cmuxterm.app.tests.tunnel"
     private let networkExtension = CloudTunnelBackend.networkExtension(extensionBundleIdentifier: CloudTunnelBannerTests.extensionID)
 
+    @Test("VPN-off guidance distinguishes cmux forwarding from other-app access")
+    func vpnOffGuidanceExplainsOptionalSystemVPN() throws {
+        let warning = try #require(CloudPortsVPNWarning.projection(tunnelState: .off))
+        #expect(warning.help.contains("in-app forwarding"))
+        #expect(warning.help.contains("Safari, Chrome"))
```
[the remark this answers is not in the document]
Comment (austinywang) [reply cmuxTests/CloudTunnelBannerTests.swift]: Fixed in b7cadb5bca. The help assertion validates both nonempty localized values and their order without relying on English substrings or changing process-wide locale. All 7 banner/help tests passed: https://github.com/manaflow-ai/cmux/actions/runs/34754539608. Both new UI strings have translations in all 20 catalog languages; the required nine-language audit passes.

File: Sources/Cloud/CloudTreeOutlineView+CloudVPN.swift [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    }
+}
+
+
+extension CloudTreeOutlineView.Coordinator {
+    func outlineViewColumnDidResize(_ notification: Notification) {
+        guard showsCloudVPNWarning, let outlineView else { return }
+        let rows = IndexSet((0..<outlineView.numberOfRows).filter {
+            (outlineView.item(atRow: $0) as? CloudTreeNode)?.isPortsEmptyPlaceholder == true
+        })
```
[the remark this answers is not in the document]
Comment (austinywang) [reply Sources/Cloud/CloudTreeOutlineView+CloudVPN.swift]: Fixed in b7cadb5bca. Snapshot application caches successful-empty-Ports nodes, using retained nodes after adopt(from:) for content-only updates. Resize resolves only those identities with row(forItem:) and discards hidden rows. All 3 resize tests passed: https://github.com/manaflow-ai/cmux/actions/runs/34754539463. The pre-fix checkpoint reached the intended assertions, including 5,005 item scans for five resizes with 1,000 unrelated rows: https://github.com/manaflow-ai/cmux/actions/runs/34754522619. It includes the prerequisite main-branch test-compilation repairs so the regression runs rather than stopping at unrelated compiler errors.

File: Sources/Cloud/CloudTreeVPNSetupButton.swift [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        return true
+    }
+
+    override func keyDown(with event: NSEvent) {
+        if [UInt16(36), 76, 49].contains(event.keyCode) { performClick(nil) }
+        else { super.keyDown(with: event) }
+    }
+
+    @objc private func invokeAction() {
+        actionHandler?()
```
[the remark this answers is not in the document]
Comment (austinywang) [reply Sources/Cloud/CloudTreeVPNSetupButton.swift]: Fixed in b7cadb5bca. CloudVPNSetupButton owns target/action, keyboard, accessibility press, enabled state, and originating-window routing for text and help-icon modes. Both presentations receive the same MachineRowActions.setupVPN closure and preserve their styling and accessibility identifiers. All 10 affordance/routing tests passed on c6b380de94, with no warnings in changed Swift files: https://github.com/manaflow-ai/cmux/actions/runs/34755825487. No empty deinit is needed: the control owns no teardown resources, as confirmed in this thread.

**Comment by austinywang** [comment]
Review follow-up on `b7cadb5bca`:

- The resize scan and duplicate native activation are fixed, and the help assertion no longer depends on the host language. Each inline thread has implementation and hosted-test links.
- Both new localization keys now have real translations in all 20 catalog languages. The repository’s required nine-language audit passes.
- Merging current `main` removed the obsolete terminal I/O, lifecycle, private-URL extraction, and Dock-fixture changes from the final PR diff.
- The Safari/Chrome examples are retained because issue #12489 explicitly requests explaining access from those browsers and other apps. These are user-facing app examples, not upstream infrastructure names or leaked error details.
- I kept documentation for the non-obvious snapshot identity, native activation, and setup-navigation contracts. I did not add empty teardown methods or boilerplate comments to trivial overrides; the new control has no teardown resources, and the repository’s architecture guidance does not impose a blanket app-target docstring percentage.

The tagged build and focused tests are running on the pushed commit. The PR remains open and will not be merged by this task.
